include/swift/SIL/SILBuilder.h

```diff
@@ -526,8 +575,11 @@ class SILBuilder {
 
   UncheckedRefCastInst *createUncheckedRefCast(SILLocation Loc, SILValue Op,
                                                SILType Ty) {
-    return insert(new (F.getModule()) UncheckedRefCastInst(
+    auto *I = insert(new (F.getModule()) UncheckedRefCastInst(
         getSILDebugLocation(Loc), Op, Ty));
+    I->getTypeDefOperandsBuf()[0].set(
+        getOpenedArchetypeDef(Ty, getModule(), OpenedArchetypes));
+    return I;
```

Comment:
Yeah, I think that's the right way to go.

Note that if you're going to collect opened archetypes, you should probably collect them in a uniquing data structure like a SetVector.